src/anomalib/metrics/__init__.py

```diff
@@ -171,25 +172,31 @@ def create_metric_collection(
 
         - if list[str] (names of metrics): see `metric_collection_from_names`
         - if dict[str, dict[str, Any]] (path and init args of a class): see `metric_collection_from_dicts`
+        - if list[Metric] (metric objects): A collection is returned with those metrics.
 
     The function will first try to retrieve the metric from the metrics defined in Anomalib metrics module,
     then in TorchMetrics package.
 
     Args:
-        metrics (list[str] | dict[str, dict[str, Any]]): List of metrics or dictionaries to create metric collection.
+        metrics (list[str] | dict[str, dict[str, Any]] | Metric | list[Metric]): List of metrics or dictionaries to
+            create metric collection.
         prefix (str | None): Prefix to assign to the metrics in the collection.
 
     Returns:
         AnomalibMetricCollection: Collection of metrics.
     """
-    # fallback is using the names
-
     if isinstance(metrics, ListConfig | list):
-        if not all(isinstance(metric, str) for metric in metrics):
-            msg = f"All metrics must be strings, found {metrics}"
+        if not (
+            all(isinstance(metric, str) for metric in metrics) or all(isinstance(metric, Metric) for metric in metrics)
+        ):
+            msg = f"All metrics must be either string or Metric objects, found {metrics}"
```

Comment:
Would this mean that a user cannot pass the following:

```python 
from torchmetrics.classification import Accuracy, Precision, Recall

from anomalib.data import MVTec
from anomalib.engine import Engine
from anomalib.models import Padim

if __name__ == "__main__":
    model = Padim()
    data = MVTec()
    engine = Engine(image_metrics=["F1Score", Accuracy(task="binary")])
    engine.train(model, datamodule=data)
```

Reply:
Also, would it be an idea to have an additional check like; 

```python
from torchmetrics.classification import Accuracy, Precision, Recall
import types

def instantiate_if_needed(metric, task="binary"):
    if isinstance(metric, types.FunctionType) or isinstance(metric, type):
        # If metric is a function or a class (not instantiated)
        return metric(task=task)
    else:
        # If metric is already instantiated
        return metric
```

Or do you think if this is overkill?